src/vector.js

```diff
@@ -53,6 +154,17 @@ Vector.cross  = function (v1, v2)
     ];
 };
 
+// Dot product of two vectors
+Vector.dot = function (v1, v2)
+{
+    var n = 0.0;
+    var lim = Math.min(v1.length, v2.length);
```

Comment:
While in some ways better to check min length of each argument (avoids any invalid accesses), it's also kind of weird, because it's not really valid to call this with arguments of mismatching length anyway, is it? For functions that might be called frequently in inner loops, we would often not carefully check arguments, but instead assume that the calling code is going to call it correctly, and just let the function throw an error if there is an invalid access. So in that case you could just use the length of the first argument and assume they are the same. I think the same may apply to some other functions above where you're checking the min of multiple arguments too?